internal/configs/virtualserver.go

```diff
@@ -490,169 +546,200 @@ type policiesCfg struct {
 	ErrorReturn     *version2.Return
 }
 
-// TODO refactor generatePolicies
-func (vsc *virtualServerConfigurator) generatePolicies(owner runtime.Object, ownerNamespace string, vsNamespace string,
-	vsName string, policyRefs []conf_v1.PolicyReference, policies map[string]*conf_v1alpha1.Policy, jwtKeys map[string]string, ingressMTLSPemFileName string, context string, tlsPemFileName string, egressMTLSSecrets map[string]string) policiesCfg {
-	var policyErrorReturn *version2.Return
-	var allow, deny []string
-	var limitReqOptions version2.LimitReqOptions
-	var limitReqZones []version2.LimitReqZone
-	var limitReqs []version2.LimitReq
-	var JWTAuth *version2.JWTAuth
-	var ingressMTLS *version2.IngressMTLS
-	var egressMTLS *version2.EgressMTLS
-	var policyError bool
+func newPoliciesConfig() *policiesCfg {
+	return &policiesCfg{}
+}
 
-	for _, p := range policyRefs {
-		polNamespace := p.Namespace
-		if polNamespace == "" {
-			polNamespace = ownerNamespace
+type policyOwnerDetails struct {
+	owner          runtime.Object
+	ownerNamespace string
+	vsNamespace    string
+	vsName         string
+}
+
+type policyOptions struct {
+	jwtKeys                map[string]string
+	ingressMTLSPemFileName string
+	tlsPemFileName         string
+	egressMTLSSecrets      map[string]string
+}
+
+type validationResults struct {
+	isError  bool
+	warnings []string
+}
+
+func newValidationResults() *validationResults {
+	return &validationResults{}
+}
+
+func (r *validationResults) addWarning(msgFmt string, args ...interface{}) {
```

Comment:
The names of the other similar functions that take a format string as a first argument end with "f". I suggest naming this function similarly:

```suggestion
func (r *validationResults) addWarningf(msgFmt string, args ...interface{}) {
```

Reply:
Thanks for that!